### PR TITLE
[#9536] Admin Search Page E2E Tests

### PR DESCRIPTION
**Pull request by jtankw3**
Part of #9536

**Outline of Solution**

- Added putDocuments method to BackDoor for BaseTestCaseWithBackDoorApiAccess to put searchable documents into the datastore
- Add AdminSearchPage which focuses on retrieving data from specific fields in the UI for comparison
- Add AdminSearchPageE2ETest which compares the main fields in the search table to the data placed via backdoor.

### CODE REVIEW

**Comment by jtankw3** [comment]
Hi @wkurniawan07, could I get a review for this PR?

File: src/e2e/java/teammates/e2e/cases/e2e/AdminSearchPageE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        searchPage = AppPage.getNewPageInstance(browser, url, AdminSearchPage.class);
+
+        browser.waitForPageLoad();
+        StudentAttributes student = testData.students.get("student1InCourse1");
+        AccountAttributes studentAccount = testData.accounts.get("student1InCourse1");
+        InstructorAttributes instructor = testData.instructors.get("instructor1OfCourse1");
+        AccountAttributes instructorAccount = testData.accounts.get("instructor1OfCourse1");
+
+        ______TS("Typical case: Search student keyword");
+        String searchContent = "Student1";
```
Comment (wkurniawan07) [inline src/e2e/java/teammates/e2e/cases/e2e/AdminSearchPageE2ETest.java]: `Student1` is a rather generic term. This may result in unexpected results when there are other data in the system.

File: src/e2e/java/teammates/e2e/cases/e2e/AdminSearchPageE2ETest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        searchPage.inputSearchContent(searchContent);
+        searchPage.clickSearchButton();
+        verifyInstructorRowContent(instructor, instructorAccount);
+        verifyInstructorExpandedLinks(instructor);
+
+        ______TS("Typical case: Reset instructor google id");
+        searchPage.resetInstructorGoogleId(instructor);
+        instructor.googleId = null;
+        verifyInstructorRowContent(instructor, instructorAccount);
+    }
```
Comment (wkurniawan07) [inline src/e2e/java/teammates/e2e/cases/e2e/AdminSearchPageE2ETest.java]: May need more example cases. I don't think resetting instructor google ID is the only action possible from this page.

File: src/main/java/teammates/ui/webapi/action/SearchAccountsAction.java [hunk 3]
```diff
@@ -267,7 +267,7 @@ private String generateResultUrl(StudentAttributes student, String fsName) {
             if (ib.getGoogleId() != null) {
                 ib.setHomePageLink(Config.getFrontEndAppUrl(Const.WebPageURIs.INSTRUCTOR_HOME_PAGE)
                         .withUserId(instructor.googleId)
-                        .toAbsoluteString());
+                        .toString());
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/webapi/action/SearchAccountsAction.java]: Reason for this?

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.html [hunk 4]
```diff
@@ -92,7 +92,7 @@
         </tr>
       </thead>
       <tbody>
-        <ng-container *ngFor="let student of students">
+        <ng-container *ngFor="let student of students;">
```
Comment (wkurniawan07) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.html]: Unrelated change; although this doesn't affect correctness, try not to have this kind of stuff.

File: src/main/java/teammates/ui/webapi/action/SearchAccountsAction.java (hunk 3 above)
Comment (jtankw3) [reply src/main/java/teammates/ui/webapi/action/SearchAccountsAction.java]: As the instructor manage account link and the student manage account link and homepage link all use relative url, I believe it would be more consistent to use a relative url for the instructor homepage link. 

It is also more convenient for testing as you would not be redirected to localhost:4200 when accessing from localhost:8080

File: src/e2e/java/teammates/e2e/cases/e2e/AdminSearchPageE2ETest.java (hunk 2 above)
Comment (jtankw3) [reply src/e2e/java/teammates/e2e/cases/e2e/AdminSearchPageE2ETest.java]: From the admin-search-page-component.ts I believe the only actions for this page are show links, hide links and reset student and instructor google id. There is also a manage account link and a homepage link which I checked but did not click. 

I will add an example case for reset student id. Is it recommended to test for hide links and correct redirection using the manage account link and a homepage link in the e2e test? Would you also recommend testing searching using different fields such as google id and course id? I am worried it may be redundant.
